src/lib/operators/table_scan/column_vs_value_table_scan_impl.cpp

```diff
@@ -127,6 +133,183 @@ void ColumnVsValueTableScanImpl::_scan_dictionary_segment(const BaseDictionarySe
+      Fail("Expected ReferenceSegments to be handled before calling this method");
+    } else {
+      // TODO(johannes-schneider): extract because of code duplication
+      // early outs for dictionary segments
+      // TODO(cmfcmf): This breaks some of our tests.
+      /*
+      if (const auto* dictionary_segment = dynamic_cast<const BaseDictionarySegment*>(&segment)) {
+        const auto search_value_id = _get_search_value_id(*dictionary_segment);
+        auto iterable = create_iterable_from_attribute_vector(*dictionary_segment);
+        if (_value_matches_all(*dictionary_segment, search_value_id)) {
+          iterable.with_iterators(position_filter, [&](auto begin, auto end) {
+            static const auto always_true = [](const auto&) { return true; };
+            _scan_with_iterators<true>(always_true, begin, end, chunk_id, matches);
+          });
+          return;
+        }
+        if (_value_matches_none(*dictionary_segment, search_value_id)) {
+          return;
+        }
+      }*/
+
+      Assert(segment.sort_order().value() == OrderByMode::AscendingNullsLast ||
+                 segment.sort_order().value() == OrderByMode::Ascending ||
+                 segment.sort_order().value() == OrderByMode::DescendingNullsLast ||
+                 segment.sort_order().value() == OrderByMode::Descending,
+             "Unsupported sort type");
+
+      // TODO(hendraet): Support Null values correctly
+      auto segment_iterable = create_iterable_from_segment(typed_segment);
+      segment_iterable.with_iterators(position_filter, [&](auto begin, auto end) {
+        auto bounds = get_sorted_bounds(position_filter, begin, end, typed_segment);
+        auto lower_it = std::get<0>(bounds);
+        auto upper_it = std::get<1>(bounds);
+        auto exclude_range = std::get<2>(bounds);
+
+        // const auto non_null_begin = segment.get_non_null_begin_offset(position_filter);
+        // const auto non_null_end = segment.get_non_null_end_offset(position_filter);
+
+        //         std::cout << "Segment contains " << std::distance(begin, end) << " elements." << std::endl
+        //                   << "Bounds contain " << std::distance(lower_it, upper_it) << " elements, start at "
+        //                   << std::distance(begin, lower_it) << " and end at " << std::distance(begin, upper_it)
+        //                   << std::endl
+        //                   << "However, the first non-null value is at " << non_null_begin
+        //                   << " and the last at " << non_null_end
+        //                   << std::endl;
+
+        size_t output_idx = matches.size();
+
+        if (exclude_range) {
+          const auto non_null_begin = segment.get_non_null_begin_offset(position_filter);
+          const auto non_null_end = segment.get_non_null_end_offset(position_filter);
+
+          const auto tmp = std::distance(upper_it, end) - (std::distance(begin, end) - non_null_end);
```

Comment:
A more expressive identifier (e.g., `dist_upper_to_null_begin`) could help to understand what distance is calculated here.